### PR TITLE
Run etcd 2.0.5 in a pod

### PR DESCRIPTION
**Pull request by ArtfulCoder**
@zmerlynn 

This PR should work correctly since we added some robustness to kube-addons creation logic with the retries #6200

### CODE REVIEW

File: cluster/saltbase/salt/etcd/etcd.manifest [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+"apiVersion": "v1beta3",
+"kind": "Pod",
+"metadata": {"name":"etcd-server"},
+"spec":{
+"hostNetwork": true,
+"containers":[
+    {
+    "name": "etcd-container",
+    "image": "kubernetes/etcd:2.0.5.1",
```
Comment (zmerlynn) [inline cluster/saltbase/salt/etcd/etcd.manifest]: Is this image on ContainerVM now, or should we get it onto gcr.io?

**Comment by zmerlynn** [comment]
LGTM except that question.

File: cluster/saltbase/salt/etcd/etcd.manifest (hunk 1 above)
Comment (dchen1107) [inline cluster/saltbase/salt/etcd/etcd.manifest]: This shouldn't be in ContainerVM image since we still support other internal users.

Comment (zmerlynn) [inline cluster/saltbase/salt/etcd/etcd.manifest]: I'm okay with merge as-is so we can start to get some run-time. Can you file a follow-on issue to move this to gcr.io quickly, though, so all infrastructure is off docker hub?

**Comment by dchen1107** [comment]
LGTM. Can you run e2e test against this change?

**Comment by dchen1107** [comment]
This is revert-of-revert-of-#4442

**Comment by dchen1107** [comment]
Please run e2e tests and post the result here. Network test was failed due to 

1) https://github.com/GoogleCloudPlatform/kubernetes/issues/5629#issuecomment-88374468
which is fixed in test itself.

2) apiserver is not response

**Comment by ArtfulCoder** [comment]
Summarizing 5 Failures:

Fail] 90mPD It] should schedule a pod w/ a RW PD, remove it, then schedule it on another host 
37m/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pd.go:107

(Failed to delete PD)

Fail] 90mkubectl update-demo It] should scale a replication controller 
37m/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/util.go:264

Fail] 90mDensity It] should allow starting 30 pods per node 
37m/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:175

Apparently pods didnt start up



Fail] 90mShell It] tests that services.sh passes 
37m/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/shell.go:67

Fail] 90mSecrets It] should be consumable from pods 
37m/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/secrets.go:111
Failed, waiting for pods to come up.


Ran 28 of 32 Specs in 5036.928 seconds
FAIL! -- 32m23 Passed | 5 Failed | 1 Pending | 3 Skipped

**Comment by ArtfulCoder** [comment]
My second test run finished
There were two failures, both related to PD tests
[It] verify monitoring pods and all cluster nodes are available on influxdb using heapster.

INFO: failed to query list of pods from influxdb. Query: "select distinct(pod_id) from /cpu.*/", Err: expected only one series from Influxdb for query "select distinct(pod_id) from /cpu.*/". Got []

 
[It] should schedule a pod w/ a RW PD, remove it, then schedule it on another host
INFO: Couldn't delete PD. Sleeping 5 seconds

**Comment by ArtfulCoder** [comment]
of the two, the first is failing on HEAD as well.

The second is failing on a "gcloud compute disks delete" command even after though the pod holding the pd has been killed. 

I believe these are not related to my change.
So we should be good to go to merge this change.

**Comment by timothysc** [comment]
@eparis ^ FYI.

**Comment by zmerlynn** [comment]
Merging to get some soak time in Jenkins.

**Comment by dchen1107** [comment]
Great!

**Comment by eparis** [comment]
@aveshagarwal

Can you check out how he docker-i-fied etcd and see if there is anything here we want to grab/contribute?

**Comment by aveshagarwal** [comment]
sure

**Pull request merged**